File: 8/task.ex
defmodule Task do

  def f_to_c(count, values) do
    f_to_c(count, values, [])
  end

  defp f_to_c(_, [], result) do
    Enum.reverse(result)
  end

  defp f_to_c(count, [head_value|tail_values], result) do
    f_to_c(count - 1, tail_values, [Kernel.round((5/9) * (head_value - 32))|result])
  end
  
  def c_to_f(count, values) do
    c_to_f(count, values, [])
  end

  defp c_to_f(_, [], result) do
    Enum.reverse(result)
  end

  defp c_to_f(count, [head_value|tail_values], result) do
    c_to_f(count - 1, tail_values, [Kernel.round(((9/5) * head_value) + 32)|result])
  end

end

File: 8-fahrenheit_to_celsius/task.erl
-module(task).

-export([f_to_c/2, c_to_f/2]).

f_to_c(Count, Values) ->
  f_to_c(Count, Values, []).

f_to_c(_, [], Result) ->
  lists:reverse(Result);
f_to_c(Count, [ValueHead|ValuesTail], Result) ->
  f_to_c(Count - 1, ValuesTail, [erlang:round((5/9) * (ValueHead - 32))|Result]).

c_to_f(Count, Values) ->
  c_to_f(Count, Values, []).

c_to_f(_, [], Result) ->
  lists:reverse(Result);
c_to_f(Count, [ValueHead|ValuesTail], Result) ->
  c_to_f(Count - 1, ValuesTail, [erlang:round(((9/5) * ValueHead) + 32)|Result]).
